### PR TITLE
Fix a concurrency issue in profileBuilder

### PR DESCRIPTION
**Pull request by ArcturusZhang**
limit the maximum concurrency to 8 by default, add a flag to manually set this

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: tools/profileBuilder/cmd/root.go [hunk 1]
```diff
@@ -62,6 +63,7 @@ func init() {
 	rootCmd.PersistentFlags().BoolVarP(&modulesFlag, "modules", "m", false, "Executes commands in modules-aware mode.")
 	rootCmd.PersistentFlags().StringVarP(&profileName, "name", "n", "", "The name that should be used to identify the profile.")
 	rootCmd.PersistentFlags().StringVarP(&outputRootDir, "output-location", "o", "", "The folder in which to output the generated profile.")
+	rootCmd.PersistentFlags().IntVar(&semLimit, "sem-limit", 8, "The limit of the maximum goroutines")
```
Comment (jhendrixMSFT) [inline tools/profileBuilder/cmd/root.go:66]: Can we not make this a flag?  More flags make the tool confusing plus is anybody really going to change this?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
What is the issue you're seeing exactly?  If there's still a race condition, lowering the number of concurrent calls will just hide it.

**Comment by jhendrixMSFT** [comment]
Ok just saw the email about this.

File: tools/profileBuilder/cmd/root.go (hunk 1 above)
Comment (jhendrixMSFT) [reply tools/profileBuilder/cmd/root.go:66]: Ok we can leave this, but perhaps a better description.  "The maximum number of packages to concurrently build." or something

**Review by WodansSon — APPROVED** [review]
This LGTM, just need to update the description of the `semLimit` flag as suggested by @jhendrixMSFT and I think this will be good to merge. 🚀

File: tools/profileBuilder/cmd/root.go (hunk 1 above)
Comment (ArcturusZhang) [reply tools/profileBuilder/cmd/root.go]: Done

**Comment by ArcturusZhang** [comment]
Hey @jhendrixMSFT I just updated the description, please take a look

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**